RevenueCatUI/CustomerCenter/ManageSubscriptionsButtonStyle.swift

```diff
@@ -17,42 +17,28 @@ import Foundation
 import RevenueCat
 import SwiftUI
 
+#if os(iOS)
+
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 @available(macOS, unavailable)
 @available(tvOS, unavailable)
 @available(watchOS, unavailable)
-struct ManageSubscriptionsButtonStyle: ButtonStyle {
+struct ManageSubscriptionsButtonStyle: PrimitiveButtonStyle {
 
     @Environment(\.appearance) private var appearance: CustomerCenterConfigData.Appearance
     @Environment(\.colorScheme) private var colorScheme
 
-    func makeBody(configuration: ButtonStyleConfiguration) -> some View {
-        configuration.label
-            .padding()
-            .frame(width: 300)
-            .background(color(from: appearance))
-            .foregroundColor(colorScheme == .dark ? Color.black : Color.white)
-            .cornerRadius(10)
-            .scaleEffect(configuration.isPressed ? 0.95 : 1.0)
-            .opacity(configuration.isPressed ? 0.8 : 1.0)
-            .animation(.easeInOut(duration: 0.2), value: configuration.isPressed)
-    }
-
-}
-
-@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
-@available(macOS, unavailable)
-@available(tvOS, unavailable)
-@available(watchOS, unavailable)
-private extension ManageSubscriptionsButtonStyle {
+    func makeBody(configuration: PrimitiveButtonStyleConfiguration) -> some View {
+        let background = color(from: appearance.buttonBackgroundColor, for: colorScheme)
+        let textColor = color(from: appearance.buttonTextColor, for: colorScheme)
 
-    func color(from appearance: CustomerCenterConfigData.Appearance) -> Color {
-        switch appearance.mode {
-        case .system:
-            return Color.accentColor
-        case .custom(accentColor: let accentColor, backgroundColor: _, textColor: _):
-            return colorScheme == .dark ? accentColor.dark.underlyingColor : accentColor.light.underlyingColor
-        }
+        Button(action: { configuration.trigger() }, label: {
+            configuration.label.frame(width: 300)
+        })
+        .buttonStyle(.borderedProminent)
+        .controlSize(.large)
+        .applyIf(background != nil, apply: { $0.tint(background) })
+        .applyIf(textColor != nil, apply: { $0.foregroundColor(textColor) })
```

Comment:
I see we're just not applying custom colors if they are null... But I'm wondering if we should customize the default style of these to be the accentColor or something like that... But I'm ok going with just the defaults for now.

Reply:
That's actually done by `.boderedProminent` and it's the reason why I am using that style by default, otherwise it would just look like text. So I think we are covered